diffsync/store/__init__.py

```diff
@@ -82,7 +86,23 @@ def remove(self, *, obj: "DiffSyncModel", remove_children: bool = False):
         Raises:
             ObjectNotFound: if the object is not present
         """
-        raise NotImplementedError
+        modelname = obj.get_type()
+        uid = obj.get_unique_id()
+
+        self._remove_item(modelname, uid)
+
+        if obj.diffsync:
+            obj.diffsync = None
+
+        if remove_children:
+            for child_type, child_fieldname in obj.get_children_mapping().items():
+                for child_id in getattr(obj, child_fieldname):
+                    try:
+                        child_obj = self.get(model=child_type, identifier=child_id)
+                        self.remove(obj=child_obj, remove_children=remove_children)
+                    except ObjectNotFound:
+                        # Since this is "cleanup" code, log an error and continue, instead of letting the exception raise
+                        self._log.error(f"Unable to remove child {child_id} of {modelname} {uid} - not found!")
```

Comment:
It occurs to me we should probably be taking advantage of structlog better on this error message. Perhaps something like:

```suggestion
                        self._log.error(f"Unable to remove child element as it was not found!", child_type=child_type, child_id=child_id, parent_type=modelname, parent_id=uid)
```